src/Reflection/Deprecation/ClassConstantDeprecationProvider.php

```diff
@@ -0,0 +1,27 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection\Deprecation;
+
+use PHPStan\BetterReflection\Reflection\Adapter\ReflectionClassConstant;
+
+/**
+ * This interface allows you to provide custom deprecation information
+ *
+ * To register it in the configuration file use the following tag:
+ *
+ * ```
+ * services:
+ * 	-
+ *		class: App\PHPStan\MyProvider
+ *		tags:
+ *			- phpstan.classConstantDeprecationProvider
+ * ```
+ *
+ * @api
+ */
+interface ClassConstantDeprecationProvider
```

Comment:
A note on naming - this has to be an "extension", not a "provider". All interfaces users are supposed to implement are called "Extension".

"Provider" is typically what allows access to those extensions in PHPStan's codebase, and it's just an internal interface to PHPStan, not covered by `@api`.

See: https://github.com/phpstan/phpstan-src/blob/2.1.x/src/DependencyInjection/Type/ParameterClosureTypeExtensionProvider.php as opposed to to https://github.com/phpstan/phpstan-src/blob/2.1.x/src/Type/FunctionParameterClosureTypeExtension.php.

So please, use PhpStorm rename refactoring to change names of those "provider" interfaces to "extension". And rename the service tags as well :)